src/SweetPoll.elm

```diff
@@ -223,13 +223,41 @@ runPoll (Model model) =
 
 toTask : String -> Decoder a -> Task Http.Error a
 toTask url decoder =
-    { method = "GET"
-    , headers = [ Http.header "Accept" "application/json" ]
-    , url = url
-    , body = Http.emptyBody
-    , expect = Http.expectJson decoder
-    , timeout = Nothing
-    , withCredentials = False
-    }
-        |> Http.request
-        |> Http.toTask
+    Http.task
+        { method = "GET"
+        , headers = [ Http.header "Accept" "application/json" ]
+        , url = url
+        , body = Http.emptyBody
+        , resolver = resolveJsonResponse decoder
+        , timeout = Nothing
+        }
+
+
+{-| `Http.task` requires you to specify your own "resolver". We don't
+actually want to do anything special here, so this just attempts to
+re-implement the expected behavior from a normal request.
+
+See `elm/http` docs [here.](https://package.elm-lang.org/packages/elm/http/latest/Http#expectStringResponse)
+
+-}
+resolveJsonResponse : Decoder a -> Http.Resolver Http.Error a
+resolveJsonResponse decoder =
```

Comment:
I think maybe this?  https://package.elm-lang.org/packages/lukewestby/http-extra/latest/Http-Extra#expectJsonResponse

Reply:
But also, it's probably not good to have this dependency have a bunch of dependencies of it's own.   I think I'm maybe prefer just having this boilerplate stuff here.

Reply:
It's pretty close, but it needs to return an `Http.Resolver e a`. Also, seems like the perfect package to add it to.